File: 백준/Silver/2559. 수열/수열.java
import java.io.*;
import java.util.*;

public class Main{

    // 필요변수
    static int N;
    static int K;
    static int[] A;

    // 입력
    static void input() throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        // 값 저장
        N = Integer.parseInt(st.nextToken());
        K = Integer.parseInt(st.nextToken());

        // 배열 선언
        A = new int[N];

        st = new StringTokenizer(br.readLine());
        // 배열 값 저장
        for (int i = 0; i < N; i++) {
            A[i] = Integer.parseInt(st.nextToken());
        }


        br.close();
    }

    // 풀이
    static void pro() {
        int sum = 0;
        int left = 0, right = 0;

        // 첫 K개 합 구하기
        for (; right < K; right++) {
            sum += A[right];
        }
        // 정답값 선언
        int maxSum = sum;

        // left는 제거하고 right는 더하기
        while (right < N) {
            sum -= A[left++];
            sum += A[right++];
            maxSum = Math.max(maxSum, sum);
        }

        System.out.println(maxSum);
        
    }
    public static void main(String[] args) throws Exception{
        input();
        pro();
    }
}

File: 백준/Silver/2559. 수열/수열.cc
#include <iostream>
using namespace std;

int N, K;
int A[100005];

void input() {
    cin >> N >> K;

    for (int i = 0; i < N; i++) {
        cin >> A[i];
    }
}


void pro() {
    int sum = 0;

    // 최초값 세팅
    for (int i = 0; i < K; i++) {
        sum += A[0 + i];
    }

    // 최초값으로 정답값 설정
    int maxSum = sum;

    // 한칸씩 이동하면서 값 확인하기
    for (int left = 0, right = K; right < N; left++, right++) {
        sum += A[right] - A[left];
        maxSum = max(sum, maxSum);
    }
     
    // 정답 출력
    cout << maxSum;
}

int main() {
    // 최적화
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    input();
    pro();
}